Sources/SwiftRepo/Repository/Protocols/Query/Query.swift

```diff
@@ -145,6 +187,66 @@ public extension Query {
     }
 }
 
+public extension Query {
+    
+    @discardableResult
+    /// Conditionally perform the query if needed based on the specified strategy and the state of the store.
+    /// This `get` function is to be used when models derived from the response will be put in a separate `Store`.
+    /// - Parameters:
+    ///   - id: The query ID
+    ///   - variables: The query variables
+    ///   - store: The observable store where query results are cached
+    ///   - unmappedKey: The store key associated with this query. This is typically either the query ID or `QueryStoreKey`, depending on the granularity of storage being used.
+    ///   - valueVariablesFactory: a closure that converts the value into its associated variables
+    ///   - keyFactory: a closure that converts the query ID and variables into a store key
+    ///   - errorIntent: The error intent to apply to errors that are thrown by the query
+    ///   - strategy: The query strategy
+    ///   - willGet: A closure that will be called if and when the query is performed. This is typically the `LoadingController.loading` function.
+    /// - Returns: The value if the query was performed. Otherwise, `nil`.
+    func get<Store, Key, ModelStore>(
+        id: QueryId,
+        variables: Variables,
+        into store: Store,
+        modelStore: ModelStore,
+        keyedBy unmappedKey: Key,
+        valueVariablesFactory: ((QueryId, Variables, Value) -> Variables)?,
+        keyFactory: @escaping (QueryId, Variables) -> Key,
+        errorIntent: ErrorIntent,
+        strategy: QueryStrategy,
+        willGet: WillGet?
+    ) async throws -> Value?
+    where Value: ModelResponse,
+          Store: ObservableStore,
+          Store.Value == Value.Value,
+          Store.Key == Key,
+          Store.PublishKey == QueryId,
+          ModelStore: SwiftRepo.Store,
+          ModelStore.Value == Value.Model,
+          ModelStore.Key == Value.Model.Key
+    {
+        try await commonGet(
+            id: id,
+            variables: variables,
+            into: store,
+            key: unmappedKey,
+            valueVariablesFactory: valueVariablesFactory,
+            keyFactory: keyFactory,
+            errorIntent: errorIntent,
+            strategy: strategy,
+            willGet: willGet,
+            storeSet: { storeKey, value in
+                // For this version, we pass `value.value`, since `Store.Value == Value.Value`
+                try await store.set(key: storeKey, value: value.value)
+            },
+            modelStoreSet: { value in
+                for model in value.models {
```

Comment:
I think we may have missed something here—we need to delete records that aren't in the new result set. Wouldn't this be the place to do it? My thought is we add a `ModelStoreStrategy` enum. I'm not exactly sure what all options are needed, but here's my initial thought:

````swift
enum ModelStoreStategy {
    /// Adds or updates models. Existing models that aren't in the current result set remain untouched.
    /// Use this strategy when the result set represents an incremental update, such as new and modified records.
    case upsertAppend

    /// Adds or updates models. Removes any models that aren't in the current result set.
    /// Use this option when the result set represents the entire data set.
    case upsertTrim
}
````
````

Reply:
Good catch.